File: chap12/fig12-7-right.R
library(dplyr)
library(NipponMap)

fit <- readRDS('output/result-model12-5.RDS')
f_ms <- fit$draws('f', format='matrix')
f_est <- apply(f_ms, 2, mean)
f_est[f_est > 19] <- 19
cols <- lattice::level.colors(
  f_est, at=seq(9, 19, length=81),
  col.regions=colorRampPalette(RColorBrewer::brewer.pal(9,'Greys'))(100)
)
png('output/fig12-7-right.png', w=800, h=600)
JapanPrefMap(col=cols)
dev.off()

File: chap12/fig12-7-right.py
import pandas
import cmdstanpy
import numpy as np
from japanmap import picture
import matplotlib.pyplot as plt

d = pandas.read_csv('input/data-map-temperature.csv')
d = pandas.concat([pandas.DataFrame({'prefID':[0], 'Y':[np.mean(d.Y)]}), d]).reset_index(drop=True)
fit = cmdstanpy.from_csv('output/result-model12-5')
f_ms = fit.stan_variable(var='f')
f_est = np.mean(f_ms, axis=0)
cmap = plt.get_cmap('gray_r')
norm = plt.Normalize(vmin=d.Y.min(), vmax=d.Y.max())
fcol = lambda x: '#' + bytes(cmap(norm(x), bytes=True)[:3]).hex()
plt.colorbar(plt.cm.ScalarMappable(norm, cmap))
plt.imshow(picture(d.Y.apply(fcol)));
plt.savefig('output/fig12-7-right.py.png')
